setup.py

```diff
@@ -25,6 +28,25 @@ def is_installed(requirement):
     else:
         return True
 
+
+def is_platform_mac():
+    return sys.platform == 'darwin'
+
+
+# For mac, ensure extensions are built for macos 10.9 when compiling on a
+# 10.9 system or above, overriding distuitls behaviour which is to target
+# the version that python was built for. This may be overridden by setting
+# MACOSX_DEPLOYMENT_TARGET before calling setup.py
+# https://github.com/pandas-dev/pandas/issues/23424#issuecomment-446393981
+if is_platform_mac():
+    if 'MACOSX_DEPLOYMENT_TARGET' not in os.environ:
+        current_system = LooseVersion(platform.mac_ver()[0])
+        python_target = LooseVersion(
+            get_config_var('MACOSX_DEPLOYMENT_TARGET'))
+        if python_target < '10.9' and current_system >= '10.9':
```

Comment:
This is not good.

In [2]: '10.10' > '10.9'
Out[2]: False

Reply:
@mckelvin 

the type of `current_system` and `python_target` is `LooseVersion`

```
In [1]: from distutils.version import LooseVersion

In [2]: LooseVersion('10.10') > '10.9'
Out[2]: True
```

maybe `if LooseVersion(python_target) < '10.9' and LooseVersion(current_system) >= '10.9':` is a better one

Reply:
Can we compare using the same type? Like `python_target < LooseVersion('10.9')`?